jplag.endToEndTesting/src/main/java/de/jplag/endToEndTesting/helper/JPlagTestSuiteHelper.java

```diff
@@ -0,0 +1,148 @@
+package de.jplag.endToEndTesting.helper;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+import java.util.Arrays;
+
+import javax.annotation.processing.FilerException;
+
+import de.jplag.endToEndTesting.constants.Constant;
+
+public class JPlagTestSuiteHelper {
+
+	private String[] resourceNames;
+	private String tempFolderPath;
+	private String[] classNames;
+
+	/**
+	 * 
+	 * @return the used temporary folder where the test data is copied to build the
+	 *         structure for the jplag api
+	 */
+	public String getFolderPath() {
+		return tempFolderPath;
+	}
+
+	/**
+	 * 
+	 * @return the class names for the current test instance that were passed in the
+	 *         constructor
+	 */
+	public String[] resourceNames() {
+		return resourceNames;
+	}
+
+	/**
+	 * Helper class for the endToEnd tests. In this class the necessary resources
+	 * are loaded, prepared and copied for the tests based on the passed parameters.
+	 * 
+	 * @param classNames of the resources that are required for the tests
+	 * @throws Exception
+	 */
+	public JPlagTestSuiteHelper(String[] classNames) throws Exception {
+		this.classNames = classNames;
+		this.resourceNames = loadResource();
+		this.tempFolderPath = getTempFolderPath();
+
+		System.out.println(String.format("temp path at [%s]", this.tempFolderPath));
+		createNewTestCaseDirectory();
+	}
+
+	/**
+	 * Copies the passed filenames to a temporary path to use them in the tests
+	 * 
+	 * @throws Exception
+	 */
+	private void createNewTestCaseDirectory() throws Exception {
+		// before copying files to the test path, check if all files are in the resource
+		// directory
+		for (String className : classNames) {
+			if (!Arrays.asList(resourceNames).contains(className)) {
+				throw new FileNotFoundException(
+						String.format("The specified class could not be found! [%s]", className));
+			}
+		}
+		// Copy the resources data to the temporary path
+		for (int counter = 0; counter < classNames.length; counter++) {
+			Path originalPath = Path.of(Constant.BASE_PATH_TO_JAVA_RESOURCES_SORTALGO.toString(), classNames[counter]);
```

Comment:
You don't need toString for Strings :)

Reply:
`Constant.BASE_PATH_TO_JAVA_RESOURCES_SORTALGO` is a path so I need that in place. Since I need a path once in the code and one time as a string I am forced to work with either `Path.of()` or `.toString()` in one of the places;
If it is a better prazis to leave the whole thing as a string in the constant file I can adjust that. Could you give me your feedback on this again? Thanks in advance